doc/Client-Portal Protocol.md

```diff
@@ -542,24 +556,124 @@ This a format for client reconnects.
        sdpMLineIndex: number(mLineIndex),   // optional in RemovedCandidatesMessage
        candidate: string(candidateSdp)
       }
+
+    object(TransportOptions)::
+      {
+        type: "webrtc" | "quic-p2p",
+        id: string(transportId) | null,  // null will result to create a new transport channel.
+      }
+
+    object(P2PQuicParametersMessage)::
+      {
+       type: "quic-parameters",
+       clientTransportParameters: object(P2PQuicClientParametersMessage) | undefined,
+       serverTransportParameters: object(P2PQuicServerParametersMessage) | undefined
+      }
+
+    object(P2PQuicClientParametersMessage)::
+      {
+        quicKey: arrayBuffer(Key),
+        iceParameters: object(IceParameters)
+      }
+    object(P2PQuicServerParametersMessage)::
+      {
+        iceParameters: object(IceParameters)
+      }
+
+    object(IceParameters)::
+      {
+        usernameFragment: string(UsernameFragment),
+        password: string(Password)
+      }
 **ResponseData**: undefined if **ResponseStatus** is “ok”.
 ### 3.3.15 Participant Receives Session Progress
 **NotificationName**: “progress”<br>
 
-**NotificationData**: The SessionProgress object with following definition.
+**NotificationData**: The SessionProgress or TransportProgress object with following definition.
+
+  object(TransportProgress)::
+    {
+      id: string(transportId),
+      status: "soac" | "ready" | "error",
+      data: object(OfferAnswer) | object(CandidateMessage) | object(P2PQuicParametersMessage) /*If status equals “soac”*/
+          | (undefined/*If status equals “ready” and session is NOT for recording*/
+          | string(Reason)/*If status equals “error”*/
+    }
 
   object(SessionProgress)::
     {
      id: string(SessionId), /* StreamId returned in publishing or SubscriptionId returned in subscribing*/
      status: "soac" | "ready" | "error",
```

Comment:
Is "soac" in "SessionProgress" still needed If the "TransportProgress" has the "soac" (offer, answer, candidate),?

Reply:
It sounds like recorder info is also soac? Should we rename it to another name?

Reply:
I think the "soac" in object(SessionProgress") can be deleted since you moved it to "TransportProgress"

Reply:
If `RecorderInfo` is removed, it's good to remove `soac` from `SessionProgress`.